### PR TITLE
[MU3 Backend] ENG-28: Combine articulations on import

### PR DESCRIPTION
**Pull request by iveshenry18**
Resolves: [ENG-28](https://mu--se.atlassian.net/jira/software/projects/ENG/boards/21?selectedIssue=ENG-28): Articulations should be combined where possible

This commit adds handling of combined articulations such as
accent staccatos and marcato tenutos on both XML import and export.
On import, pairs of articulations are combined where possible; on
export, these combined articulation symbols are separated into their
component MusicXML elements. This also adds a test for these cases,
and updates a test that previously didn't take this effect into account.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: importexport/musicxml/importmxmlpass2.cpp [hunk 1]
```diff
@@ -5432,6 +5432,9 @@ void MusicXMLParserNotations::slur()
       if (notation.attribute("type") == "stop") {
             _slurStop = true;
             }
+      if (notation.attribute("type") == "start") {
```
Comment (Jojo-Schmitz) [inline importexport/musicxml/importmxmlpass2.cpp:5435]: not an `else if` here?

Comment (iveshenry18) [reply importexport/musicxml/importmxmlpass2.cpp:5435]: Good call; I'll go ahead and make that change!

**Review by oktophonie — APPROVED** [review]

File: importexport/musicxml/importmxmlpass2.cpp [hunk 2]
```diff
@@ -6110,6 +6113,35 @@ Notation Notation::notationWithAttributes(const QString& name, const QXmlStreamA
...
+      std::vector<QString> names{ n1.name(), n2.name() };
+      std::sort(names.begin(), names.end());
+      QString name = names[0] + " " + names[1];
+
+      // Parents should match (and will both be "articulation")
+      Q_ASSERT(n1.parent() == n2.parent());
+      QString parent = n1.parent();
+
+      Notation mergedNotation { name, parent, symId };
+      for (const std::pair<const QString, QString>& attr : n1.attributes()) {
```
Comment (vpereverzev) [inline importexport/musicxml/importmxmlpass2.cpp:6136]: prefer to use `const auto& attr` in such cases

File: importexport/musicxml/importmxmlpass2.cpp [hunk 3]
```diff
@@ -6119,6 +6151,15 @@ void Notation::addAttribute(const QStringRef name, const QStringRef value)
       _attributes.insert(std::pair<QString, QString>(name.toString(), value.toString()));
       }
 
+//---------------------------------------------------------
+//   addAttribute
+//---------------------------------------------------------
+
+void Notation::addAttribute(const QString& name, const QString& value)
+      {
+      _attributes.insert(std::pair<QString, QString>(name, value));
```
Comment (vpereverzev) [inline importexport/musicxml/importmxmlpass2.cpp:6160]: I'd suggest to use emplace in this case
_attributes.emplace(name, value);

**Comment by iveshenry18** [comment]
@vpereverzev I appreciate the tips! I went ahead and added those items in the last commit.

**Review by vpereverzev — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
This PR would resolve #16748

**Pull request merged**